---
title: "H3.3 ChIP-seq overlapped with RNA-seq of siH3.3"
author: "Joshua Soon"
date: '2022-18-05'
output: html_document
editor_options: 
  chunk_output_type: console
---

Load essential libraries for relative pathing and yaml parsing.
```{r echo=FALSE, warning=FALSE, message=FALSE}
library(here)
library(yaml)
```


Load the config file with data paths.
```{r echo=FALSE, warning=FALSE, message=FALSE}
config <- yaml.load_file("config.yml")
```



Load the libraries specific to this project.
```{r echo=FALSE, warning=FALSE, message=FALSE}
library(ChIPseeker)
library(clusterProfiler)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(ReactomePA)
library(DOSE)
library(org.Hs.eg.db)
```



Load data
```{r echo=FALSE, warning=FALSE, message=FALSE}
RNA_seq_data_file <- config$Paths$RNAseq_DEGs
ChIP_seq_data_file <- config$Paths$Narrow_peaks_bed

RNA_seq_data <- read.csv(RNA_seq_data_file)

narrow_peaks <- readPeakFile(ChIP_seq_data_file)
```



Load annotation database for region specific analysis of peak occupancy and peak annotations.
```{r echo=False, warning=False, message=False}
txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene
narrow_peaks_Anno <- annotatePeak(narrow_peaks, tssRegion=c(-3000, 3000), TxDb=txdb, annoDb="org.Hs.eg.db")
```



Extract entrez gene IDs from ChIP-seq data and only keep unique ones
```{r echo=False, warning=False, message=False}
ChIP_seq_entrez_IDs <- unique(as.data.frame(narrow_peaks_Anno)$geneId)
```



Extract Gene symbols from RNA-seq data and convert to entrez IDs amd keep only unique ones
```{r echo=False, warning=False, message=False}
RNA_seq_gene_symbols <- unique(RNA_seq_data$gene)
RNA_seq_entrez_IDs <- mapIds(org.Hs.eg.db, RNA_seq_gene_symbols, 'ENTREZID', 'SYMBOL')
```


Intersect genes between ChIP-seq and RNA-seq
```{r echo=False, warning=False, message=False}
common_genes <- intersect(ChIP_seq_entrez_IDs, RNA_seq_entrez_IDs)
background_genes <- union(ChIP_seq_entrez_IDs, RNA_seq_entrez_IDs)
```



```{r echo=False, warning=False, message=False}



```