---
title: "H3.3 ChIP-seq overlapped with RNA-seq of siH3.3"
author: "Joshua Soon"
date: '2022-18-05'
output: html_document
editor_options: 
  chunk_output_type: console
---

Load essential libraries for relative pathing and yaml parsing.
```{r echo=FALSE, warning=FALSE, message=FALSE}
library(here)
library(yaml)
```


Load the config file with data paths.
```{r echo=FALSE, warning=FALSE, message=FALSE}
config <- yaml.load_file("config.yml")
```



Load the libraries specific to this project.
```{r echo=FALSE, warning=FALSE, message=FALSE}
library(ChIPseeker)
library(clusterProfiler)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(ReactomePA)
library(DOSE)
library(org.Hs.eg.db)
```



Load data
```{r echo=FALSE, warning=FALSE, message=FALSE}
RNA_seq_data_file <- config$Paths$RNAseq_DEGs
ChIP_seq_data_file <- config$Paths$Narrow_peaks_bed
BETA_H3F3A_uptarget_associate_peaks <- config$Paths$BETA_H3F3A_uptarget_associate_peaks

RNA_seq_data <- read.csv(RNA_seq_data_file)

narrow_peaks <- readPeakFile(ChIP_seq_data_file)
BETA_peaks <- readPeakFile(BETA_H3F3A_uptarget_associate_peaks)
```



Heatmap of BETA detected H3F3A peaks associated with transcriptional up-regulation
```{r echo=False, warning=False, message=False}
BETA_promoters <- getPromoters(TxDb=txdb, upstream=3000, downstream=3000)
BETA_tagMatrix <- getTagMatrix(BETA_peaks, windows=promoters)

peakHeatmap(BETA_peaks, TxDb=txdb, upstream=3000, downstream=3000, color="red")
```



Load annotation database for region specific analysis of peak occupancy and peak annotations.
```{r echo=False, warning=False, message=False}
txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene
narrow_peaks_Anno <- annotatePeak(narrow_peaks, tssRegion=c(-3000, 3000), TxDb=txdb, annoDb="org.Hs.eg.db")
```



Extract entrez gene IDs from ChIP-seq data and only keep unique ones
```{r echo=False, warning=False, message=False}
ChIP_seq_entrez_IDs <- unique(as.data.frame(narrow_peaks_Anno)$geneId)
```



Extract Gene symbols from RNA-seq data and convert to entrez IDs amd keep only unique ones
```{r echo=False, warning=False, message=False}
RNA_seq_gene_symbols <- unique(RNA_seq_data$gene)
RNA_seq_entrez_IDs <- mapIds(org.Hs.eg.db, RNA_seq_gene_symbols, 'ENTREZID', 'SYMBOL')
```


Intersect genes between ChIP-seq and RNA-seq
```{r echo=False, warning=False, message=False}
common_genes <- intersect(ChIP_seq_entrez_IDs, RNA_seq_entrez_IDs)
background_genes <- union(ChIP_seq_entrez_IDs, RNA_seq_entrez_IDs)
```



Functional enrichment analysis with:
1. Reactome PA ORA
2. GO BP ORA
3. GO CC ORA
4. GO MF ORA
5. Disease gene network enrichment


1. Reactome PA ORA
```{r echo=False, warning=False, message=False}
pathways <- enrichPathway(gene = common_genes, organism = "human", universe = background_genes)

pathways_dotplot <- dotplot(pathways, showCategory=10)
pathways_dotplot

pathways_barplot <- barplot(pathways)
pathways_barplot
```



2. GO BP ORA
```{r echo=False, warning=False, message=False}
enrich_GO_BP <- enrichGO(gene = common_genes, OrgDb = org.Hs.eg.db, ont = "BP", universe = background_genes)

go_bp_dotplot <- dotplot(enrich_GO_BP, showCategory=10)
go_bp_dotplot

go_bp_barplot <- barplot(enrich_GO_BP)
go_bp_barplot
```



3. GO CC ORA
```{r echo=False, warning=False, message=False}
enrich_GO_CC <- enrichGO(gene = common_genes, OrgDb = org.Hs.eg.db, ont = "CC", universe = background_genes)

go_cc_dotplot <- dotplot(enrich_GO_CC, showCategory=10)
go_cc_dotplot

go_cc_barplot <- barplot(enrich_GO_CC)
go_cc_barplot
```



4. GO MF ORA
```{r echo=False, warning=False, message=False}
enrich_GO_MF <- enrichGO(gene = common_genes, OrgDb = org.Hs.eg.db, ont = "MF", universe = background_genes)

go_MF_dotplot <- dotplot(enrich_GO_MF, showCategory=10)
go_MF_dotplot

go_MF_barplot <- barplot(enrich_GO_MF)
go_MF_barplot
```



5. Disease gene network enrichment
```{r echo=False, warning=False, message=False}
enrich_DGN <- enrichDGN(gene = common_genes, universe = background_genes)

DGN_dotplot <- dotplot(enrich_DGN, showCategory=10)
DGN_dotplot

DGN_barplot <- barplot(enrich_DGN)
DGN_barplot
```
